pkg/controller/aws/cache/managed.go

```diff
@@ -73,7 +73,8 @@ func AddManaged(mgr manager.Manager) error {
 }
 
 type connecter struct {
-	client client.Client
+	client      client.Client
+	newClientFn func(credentials []byte, region string) (elasticache.Client, error)
```

Comment:
I wonder if we should make this an interface in the `elasticache` package. Something like:

```go
type ClientProducer interface {
    ProduceClient(credentials []byte, region string) (elasticache.Client, error)
}

type ClientProducerFn func(credentials []byte, region string) (elasticache.Client, error)

func (fn ClientProducerFn)  ProduceClient(credentials []byte, region string) (elasticache.Client, error) {
    return fn(credentials, region)
}

type connecter struct {
    kubeClient       client.Client
    ecClientProducer ClientProducer
}

c := &connecter{
    kubeClient: mgr.NewClient(),

    // This assumes we rename elasticache.NewClient to elasticache.ProduceClient
    ecClientProducer: ClientProducerFn(ProduceClient)
}
```

Not necessary given what you have here is pretty much identical to the current pattern, but it feels a little cleaner.

Reply:
If we could come up with an interface for all resources or for all let's say all aws resources, then it'd make more sense. But it's only called in one place and specific for elasticache, I'm not sure if +1 interface and type is worth it.